### PR TITLE
Fix ready state

### PR DESCRIPTION
**Pull request by mirpo**
I left some extra console.log so you could recheck me, by commenting out one line with                 this.connections.get(address).close() and run npm run network N

### CODE REVIEW

File: src/connection/WsEndpoint.js [hunk 1]
```diff
@@ -67,12 +67,38 @@ class WsEndpoint extends EventEmitter {
 
         this.wss.on('connection', this._onIncomingConnection.bind(this))
 
+        this.wss.options.verifyClient = (info) => {
+            const parameters = url.parse(info.req.url, true)
+            const { address } = parameters.query
+
+            if (this.isConnected(address)) {
```
Comment (harbu) [inline src/connection/WsEndpoint.js:74]: Interesting. Do I understand correctly that in this case you drop existing socket if a new socket is established between the same two peers?

File: src/connection/WsEndpoint.js [hunk 2]
```diff
@@ -67,12 +67,38 @@ class WsEndpoint extends EventEmitter {
...
+            return true
+        }
+
         // Attach custom headers to headers before they are sent to client
         this.wss.on('headers', (headers) => {
             headers.push(...this.customHeaders.asArray())
         })
 
         debug('listening on: %s', this.getAddress())
+        this.checkConnections = setInterval(this._checkConnections.bind(this), 2000)
```
Comment (harbu) [inline src/connection/WsEndpoint.js]: Good idea. Let's keep this to make sure that no problems with readyState in future?

File: src/connection/WsEndpoint.js (hunk 1 above)
Comment (harbu) [reply src/connection/WsEndpoint.js:74]: Do we still need the duplicate socket dropping in `this._onNewConnection`?

**Review by harbu — CHANGES_REQUESTED** [review]
tests need to be fixed; LGTM otherwise.

File: src/connection/WsEndpoint.js (hunk 1 above)
Comment (mirpo) [reply src/connection/WsEndpoint.js:74]: Yes

File: src/connection/WsEndpoint.js (hunk 2 above)
Comment (mirpo) [reply src/connection/WsEndpoint.js]: :thumbsup:

**Pull request merged**